examples/algorithms/plot_subgraphs.py

```diff
@@ -0,0 +1,170 @@
+"""
```

Comment:
Related to this PR?

Reply:
Yeah I had to merge with `main` at least twice during the development to resolve conflicts and the rebase was pretty complicated so I was lazy and just merged, relying on the final squash to clean up the history "/. So no - these changes aren't directly related.